src/org/omegat/filters3/xml/xliff/XLIFFDialect.java

```diff
@@ -160,6 +166,48 @@ public Boolean validateContentBasedTag(String tag, Attributes atts) {
         return "mrk".equals(tag) && atts != null && "protected".equals(atts.getValueByName("mtype"));
     }
 
+    /**
+     * Handle &lt;target state="needs-translated"&gt; attribute according to
+     * option settings. In default, change to &lt;target state="translated"&gt;.
+     * It also handle it when source has state="new" to "translated" or "needs-translation".
+     * @see <a href="https://sourceforge.net/p/omegat/feature-requests/1506/">RFE #1506</a>
+     * @param tag XML tag to be processed.
+     * @param translated is the value considered translated?
+     */
+    @Override
+    public void handleXMLTag(XMLTag tag, boolean translated) {
+        for (int i = 0; i < tag.getAttributes().size(); i++) {
+            Attribute attr = tag.getAttributes().get(i);
+            if (tag.getTag().equals("target") && attr.getName().equals("state")) {
+                String current = attr.getValue();
+                if ("needs-translation".equals(current)) {
+                    if (!keepStateNeedsTranslation || translated) {
+                        if (changeStateToNeedsReviewTranslation) {
+                            attr.setValue("needs-review-translation");
+                        } else {
+                            attr.setValue("translated");
+                        }
+                    } else {
```

Comment:
I kind of liked before how you had a transition table defined as a Map. Was there a reason you switch to this style?

Reply:
I've rewrite it with Map.